extra_geom/detectors.py

```diff
@@ -768,6 +769,123 @@ def offset(self, shift, *, modules=np.s_[:], tiles=np.s_[:]):
         ])
 
 
+class GenericGeometry(DetectorGeometryBase):
+    """A generic detector layout based either on the CrystFEL geom file or on a set of parameters.
+
+    The coordinates used in this class are 3D (x, y, z), and represent metres.
+
+    The :attr:`expected_data_shape` is a following triple :
+
+        1. the number of modules :attr:`n_modules`
+        2. the number of tiles in a module along the slow-scan direction multiplied by
+           the number of slow-scan pixels per tile :attr:`frag_ss_pixels`
+        3. the number of tiles in a module along the fast-scan direction multiplied by
+           the number of fast-scan pixels per tile :attr:`frag_fs_pixels`
+    """
+    detector_type_name = 'Generic Detector'
+
+    def __init__(self, pixel_size: float, fast_pixels: int, slow_pixels: int,
```

Comment:
Nitpick: I try to consistently have the slow-scan parameter before the fast-scan parameter - this matches the way you use them when indexing `module_data[slow_scan, fast_scan]`.

Reply:
I naturally sort things alphabetically but I always support any systematic approach. Done.